examples/pytorch_lightning/relational_gat.py

```diff
@@ -0,0 +1,165 @@
+
+    def val_dataloader(self):
+        return NeighborLoader(
+            self.data, num_neighbors=[10] * 2, shuffle=False, batch_size=1024,
+            input_nodes=('paper', self.data['paper'].val_mask), num_workers=6,
+            persistent_workers=True)
+
+    def test_dataloader(self):
+        return NeighborLoader(
+            self.data, num_neighbors=[10] * 2, shuffle=False, batch_size=1024,
+            input_nodes=('paper', self.data['paper'].test_mask), num_workers=6,
+            persistent_workers=True)
+
+
+class GAT(torch.nn.Module):
+    def __init__(
+        self,
+        hidden_channels: int,
+        out_channels: int,
+        dropout: float = 0.5,
+    ):
+        super().__init__()
+        self.dropout = dropout
+
+        self.conv1 = GATConv((-1, -1), hidden_channels, add_self_loops=False)
+        self.conv2 = GATConv((-1, -1), out_channels, add_self_loops=False)
+
+        self.lin1 = Linear(-1, hidden_channels)
+        self.lin2 = Linear(-1, out_channels)
+
+    def forward(self, x: Tensor, edge_index: Tensor) -> Tensor:
+        x = self.conv1(x, edge_index) + self.lin1(x)
+        x = x.relu_()
+        x = F.dropout(x, p=self.dropout, training=self.training)
+        x = self.conv2(x, edge_index) + self.lin2(x)
+        return x
+
+
+class RelationalGAT(LightningModule):
+    def __init__(
+        self,
+        hidden_channels: int,
+        out_channels: int,
+        metadata: Tuple[List[NodeType], List[EdgeType]],
+        dropout: float = 0.5,
+    ):
+        super().__init__()
+        self.save_hyperparameters()
+
+        self.model = GAT(hidden_channels, out_channels, dropout)
+        # Convert the homogeneous GAT model to a heterogeneous variant in
+        # which distinct parameters are learned for each node and edge type.
+        self.model = to_hetero(self.model, metadata, aggr='sum')
+
+        self.train_acc = Accuracy()
+        self.val_acc = Accuracy()
+        self.test_acc = Accuracy()
+
+    def forward(
+        self,
+        x_dict: Dict[NodeType, Tensor],
+        edge_index_dict: Dict[EdgeType, Tensor],
+    ) -> Dict[NodeType, Tensor]:
+        return self.model(x_dict, edge_index_dict)
+
+    @torch.no_grad()
+    def setup(self, stage: Optional[str] = None):  # Initialize parameters.
+        batch = next(iter(self.trainer.datamodule.val_dataloader()))
+        self(batch.x_dict, batch.edge_index_dict)
+
+    def common_step(self, batch: Batch) -> Tuple[Tensor, Tensor]:
+        batch_size = batch['paper'].batch_size
+        y_hat = self(batch.x_dict, batch.edge_index_dict)['paper'][:batch_size]
+        y = batch['paper'].y[:batch_size]
+        return y_hat, y
+
+    def training_step(self, batch: Batch, batch_idx: int) -> Tensor:
+        y_hat, y = self.common_step(batch)
+        loss = F.cross_entropy(y_hat, y)
+        self.train_acc(y_hat.softmax(dim=-1), y)
+        self.log('train_acc', self.train_acc, prog_bar=True, on_step=False,
```

Comment:
You might want to log on step during training, especially if there is lot of batches.